model/job.go

```diff
@@ -32,6 +33,36 @@ type Job struct {
 	Data           map[string]interface{} `json:"data"`
 }
 
+func (j *Job) IsValid() *AppError {
+	if len(j.Id) != 26 {
+		return NewAppError("Job.IsValid", "model.job.is_valid.id.app_error", nil, "id="+j.Id, http.StatusBadRequest)
+	}
+
+	if j.CreateAt == 0 {
+		return NewAppError("Job.IsValid", "model.job.is_valid.create_at.app_error", nil, "id="+j.Id, http.StatusBadRequest)
+	}
+
+	switch j.Type {
```

Comment:
I thought this looked broken at first, but I forgot Go doesn't fall through by default